Mailer/app/api/v1/Handlers/Users/UnSubscribeHandler.php

```diff
@@ -25,43 +25,37 @@ public function __construct(
         $this->listsRepository = $listsRepository;
     }
 
-    public function params()
-    {
-        return [
-            new InputParam(InputParam::TYPE_POST_RAW, 'raw')
-        ];
-    }
-
-    public function handle($params)
+    public function handle($payload)
     {
-        try {
-            $data = Json::decode($params['raw'], Json::FORCE_ARRAY);
-        } catch (JsonException $e) {
-            return new JsonApiResponse(400, ['status' => 'error', 'message' => 'input data was not valid JSON']);
-        }
+        $request = file_get_contents("php://input");
 
-        if (!isset($data['email'])) {
-            return new JsonApiResponse(400, ['status' => 'error', 'message' => 'required field missing: email']);
-        }
-        if (!isset($data['user_id'])) {
-            return new JsonApiResponse(400, ['status' => 'error', 'message' => 'required field missing: user_id']);
-        }
+        $payload = $this->validateInput($request, 'unsubscribe.schema.json');
 
-        if (!isset($data['list_id']) && !isset($data['list_code'])) {
-            return new JsonApiResponse(400, ['status' => 'error', 'message' => 'required field missing: list_id or list_code']);
+        if ($this->hasErrorResponse()){
+            return $this->getErrorResponse();
         }
 
-        if (isset($data['list_code'])) {
-            $list = $this->listsRepository->findByCode($data['list_code'])->fetch();
+        if (isset($payload['list_code'])) {
+            $list = $this->listsRepository->findByCode($payload['list_code'])->fetch();
         } else {
-            $list = $this->listsRepository->find($data['list_id']);
+            $list = $this->listsRepository->find($payload['list_id']);
         }
 
         if ($list === false) {
             return new JsonApiResponse(404, ['status' => 'error', 'message' => 'list not found']);
         }
 
-        $this->userSubscriptionsRepository->unsubscribeUser($list, $data['user_id'], $data['email'], $data['utm_params'] ?? []);
+        $userSubscription = $this->userSubscriptionsRepository->getUserSubscription($list, $payload['user_id'], $payload['email']);
```

Comment:
Fetching this `$userSubscription` only makes sense if there's a `$payload['variant_id']`, isn't it? Let's not fetch it if it's not necessary.